app/services/record_declarations/event_factory.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module RecordDeclarations
+  class EventFactory
+    class << self
+      def call(course)
+        event_namespace_for_event(course)
+      end
+
+    private
+
+      def event_namespace_for_event(course)
+        event_identifiers[course.underscore.intern].to_s
+      end
+
+      def event_identifiers
+        started_identifiers.merge(retained_identifiers)
+      end
```

Comment:
So is this equivalent to a hash like so:
```
{
  started: "Started",
  completed: "Started",
  retained_1: "Retained",
  retained_2: "Retained",
  retained_3: "Retained",
  retained_4: "Retained",
}
```

If so, can we please hard-code and inline it?

And perhaps a more important question: why is `completed` identified with `Started`?

Reply:
No. This is generated because when we change induction courses these will have different values.

Reply:
What do you mean by change induction course? What will the values be?

Reply:
Completed and Started have equivalences at the moment. We're probably going to add more specifics since the declaration types are going to change based on other factors, but at the moment, the "started" and "completed" both are the same (with the exception of the stored declaration_type). These may need to be more explicit later, but for the purposes of standing up the API to support them, this is the correct overall shape for now.

Reply:
@ethanmills .... @blackrat deleted the comment you responded to, much confusion ensues lol

Reply:
I will need to spike into a nicer way of validating api requests early next week :P 

For time being I think this is ok. `started` and `completed` are different in the database, but have the same shape for the purpose of schema validation. And this code's purpose is to find the right schema validator. We might end up renaming them to something nicer later down the line.